ec2/ec2test/server.go

```diff
@@ -1955,13 +2037,26 @@ func (srv *Server) describeIFaces(w http.ResponseWriter, req *http.Request, reqI
 	resp.XMLName = xml.Name{defaultXMLName, "DescribeNetworkInterfacesResponse"}
 	resp.RequestId = reqId
 	for _, i := range srv.ifaces {
-		ok, err := f.ok(i)
-		_, known := idMap[i.Id]
-		if ok && (len(idMap) == 0 || known) {
+		// Without neither filers nor ids - return everything.
+		if len(f) == 0 && len(idMap) == 0 {
 			resp.Interfaces = append(resp.Interfaces, i.NetworkInterface)
-		} else if err != nil {
+			continue
+		}
+		filterMatch, err := f.ok(i)
+		if err != nil {
 			fatalf(400, "InvalidParameterValue", "describe ifaces: %v", err)
 		}
+		_, requestedByID := idMap[i.Id]
+		if len(idMap) == 0 {
+			// since filter.ok() returns true when none matches, we
```

Comment:
This is confusing.

I *think* you mean that filter.ok() returns true when there are no filters specified, and so not specifying an ID means you don't implicitly filter by ID either.

I think this would be easier to understand, and would not require commentary, if you checked `len(idMap) == 0 || idMap[i.Id]` before everything else, continuing if the test fails.